Source/Core/Core/AchievementManager.cpp

```diff
@@ -5,17 +5,23 @@
 
 #include "Core/AchievementManager.h"
 
-#include <array>
 #include <rcheevos/include/rc_hash.h>
 
 #include "Common/HttpRequest.h"
 #include "Common/WorkQueueThread.h"
-#include "Config/AchievementSettings.h"
+#include "Core/Config/AchievementSettings.h"
 #include "Core/Core.h"
+#include "Core/PowerPC/MMU.h"
+#include "Core/System.h"
 #include "DiscIO/Volume.h"
+#include "VideoCommon/VideoEvents.h"
 
 static constexpr bool hardcore_mode_enabled = false;
 
+AchievementManager::AchievementManager() : m_system{Core::System::GetInstance()}
```

Comment:
Would it make more sense for System to own AchievementManager rather than vice versa? And of course going off another comment, AchievementManager should not be a singleton if it is obtaining a System reference.

Reply:
Chiming in here, this depends on what you want the lifetime of AchievementManager to be. My eventual design for System is that it is allocated on emulation start and destroyed on emulation end, so if there's no game running there's no system. If the AchievementManager should be alive outside of those times, then it needs to exist outside the System somewhere.

Reply:
AchievementManager preferably is alive when games are not running. We only allow logging in and entering hardcore mode when a game isn't running and it would be much more convenient to handle this through Dolphin's outer UI than it would be to require restarting Dolphin altogether for it.

Reply:
Makes sense, which case at least, perhaps it should not be obtaining a reference to a System but rather a pointer, so it may be invalidated/set to nullptr when the reference is no longer valid (and probably asserts placed where it's expected for the System pointer to be valid)